src/caret_analyze/architecture/struct/callback.py

```diff
@@ -112,6 +112,11 @@ def publish_topic_names(self) -> Optional[List[str]]:
     def to_value(self) -> CallbackStructValue:
         pass
 
+    def assign_publisher(self, pub_topic_name: str):
+        if self._publish_topic_names is not None and\
+           pub_topic_name not in self._publish_topic_names:
+            self._publish_topic_names.append(pub_topic_name)
```

Comment:
If none, please add pub_topic_name to a new list.
```suggestion
        self._publish_topic_names = self._publish_topic_names or []
        if pub_topic_name not in self._publish_topic_names:
            self._publish_topic_names.append(pub_topic_name)
```

Reply:
Fix it in [93f96b9](https://github.com/tier4/CARET_analyze/pull/196/commits/93f96b9a54029c50c1d9392d5c2748cfb8ab3e3c).